tokio/src/runtime/blocking/pool.rs

```diff
@@ -403,52 +399,64 @@ impl Spawner {
             return Err(SpawnError::ShuttingDown);
         }
 
-        shared.queue.push_back(task);
+        // Get thread count for adaptive sharding
+        let num_threads = self.inner.metrics.num_threads();
```

Comment:
You may want the number of worker thread instead of the number of live thread in the blocking pool.